core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -317,6 +364,197 @@ public void close() throws Exception {
         // TODO: Provide Implementation
     }
 
+    private ShareSessionKey shareSessionKey(String groupId, Uuid memberId) {
+        return new ShareSessionKey(groupId, memberId);
+    }
+
+    private static String partitionsToLogString(Collection<TopicIdPartition> partitions) {
+        return FetchSession.partitionsToLogString(partitions, log.isTraceEnabled());
+    }
+
+    /**
+     * Recursive function to process all the fetch requests present inside the fetch queue
+     */
+    private void maybeProcessFetchQueue() {
+        if (!processFetchQueueLock.compareAndSet(false, true)) {
+            // The queue is already being processed hence avoid re-triggering.
+            return;
+        }
+
+        // Initialize the topic partitions for which the fetch should be attempted.
+        Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData = new LinkedHashMap<>();
+        ShareFetchPartitionData shareFetchPartitionData = fetchQueue.poll();
```

Comment:
I'm not convinced that this will always return an entry. Adding an entry and then calling this method are not synchronized (which is fine) but I expect there is a situation in which the added entry has already been processed before this method is called.

Reply:
Added a null check, done.